WordPress/Classes/ViewRelated/Plans/PlanDetailViewController.swift

```diff
@@ -247,6 +226,65 @@ class PlanDetailViewController: UIViewController {
             alert.presentFromRootViewController()
         }
     }
+
+    // MARK: - NoResults Handling
+
+    private func setupNoResultsViewController() {
+        let noResultsStoryboard = UIStoryboard(name: "NoResults", bundle: nil)
+        guard let noResultsViewController = noResultsStoryboard.instantiateViewController(withIdentifier: "NoResults") as? NoResultsViewController else {
+            return
+        }
+
+        noResultsViewController.delegate = self
+        self.noResultsViewController = noResultsViewController
+    }
+
+    private func updateNoResults() {
+        hideNoResults()
+        if let noResultsViewModel = viewModel.noResultsViewModel {
+            showNoResults(noResultsViewModel)
+        }
+    }
+
+    private func showNoResults(_ viewModel: NoResultsViewController.Model) {
+
+        if noResultsViewController == nil {
+            setupNoResultsViewController()
+        }
+
+        guard let noResultsViewController = noResultsViewController else {
+            return
+        }
+
+        noResultsViewController.bindViewModel(viewModel)
+
+        noResultsViewController.view.frame = noResultsViewFrame()
+        noResultsViewController.view.backgroundColor = UIColor.white
+
+        tableView.addSubview(withFadeAnimation: noResultsViewController.view)
+        addChildViewController(noResultsViewController)
+
+        noResultsViewController.didMove(toParentViewController: self)
```

Comment:
This process of adding a child view controller is also something we would have to do always. Maybe we can hide it too in the `NoResultsViewController` class?
I'm not sure if the implementations will always be the same for every case but it would be cool if that's possible :)

Reply:
Hey @etoledom . You noticed the repetitiveness did ya? 😄 Yea, I did too. And that was actually going to be my next task - extract out as much of that as possible to the `NoResultsViewController` class. I just didn't make it part of this PR because I wanted to look at it further. Just as you said - not implementations will be the same.  So I wanted to flush out the patterns before doing so.

Reply:
Perfect! ✅